### PR TITLE
Bug 1924983: localize components in utils directory

### PR DESCRIPTION
**Pull request by jcaianirh**
Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1924983

### CODE REVIEW

File: frontend/public/components/utils/copy-to-clipboard.tsx [hunk 1]
```diff
@@ -29,7 +31,7 @@ export const CopyToClipboard: React.FC<CopyToClipboardProps> = React.memo((props
             type="button"
           >
             <CopyIcon />
-            <span className="sr-only">Copy to Clipboard</span>
+            <span className="sr-only">{t('utils~Copy to Clipboard')}</span>
```
Comment (rebeccaalpert) [inline frontend/public/components/utils/copy-to-clipboard.tsx:34]: I think this should be 
```suggestion
            <span className="sr-only">{t('utils~Copy to clipboard')}</span>
```

File: frontend/public/components/utils/list-input.tsx [hunk 2]
```diff
@@ -89,7 +91,7 @@ export class ListInput extends React.Component<ListInputProps, ListInputState> {
           variant="link"
         >
           <PlusCircleIcon className="co-icon-space-r" />
-          Add More
+          {t('utils~Add More')}
```
Comment (rebeccaalpert) [inline frontend/public/components/utils/list-input.tsx:94]:
```suggestion
          {t('utils~Add more')}
```

File: frontend/public/components/utils/storage-class-dropdown.tsx [hunk 3]
```diff
@@ -159,13 +165,13 @@ export class StorageClassDropdownInner extends React.Component<
               }
               htmlFor={id}
             >
-              Storage Class
+              {t('utils~Storage class')}
```
Comment (rebeccaalpert) [inline frontend/public/components/utils/storage-class-dropdown.tsx:168]: I think this is a K8sKind so it needs to be StorageClass throughout.

**Comment by jcaianirh** [comment]
@rebeccaalpert thanks for the review, completed all review requests

**Review by rebeccaalpert — APPROVED** [review]
Looks good to me! Thank you.

**Comment by jcaianirh** [comment]
@spadgett this is ready for review

File: frontend/public/components/utils/build-hooks.tsx [hunk 4]
```diff
@@ -8,24 +9,25 @@ export const BuildHooks: React.SFC<BuildHooksProps> = ({ resource }) => {
   const postCommitArgs = _.get(resource, 'spec.postCommit.args');
   const postCommitCommand = _.get(resource, 'spec.postCommit.command');
   const postCommitScript = _.get(resource, 'spec.postCommit.script');
+  const { t } = useTranslation();
 
   return !_.isEmpty(postCommitCommand) || !_.isEmpty(postCommitArgs) || postCommitScript ? (
     <div className="co-m-pane__body">
-      <SectionHeading text="Post-Commit Hooks" />
+      <SectionHeading text={t('utils~Post-Commit Hooks')} />
```
Comment (spadgett) [inline frontend/public/components/utils/build-hooks.tsx:16]:
```suggestion
      <SectionHeading text={t('utils~Post-commit hooks')} />
```

File: frontend/public/components/utils/alerts.tsx [hunk 5]
```diff
@@ -23,9 +25,9 @@ export const ExpandableAlert: React.FC<CustomAlertProps> = ({ alerts, variant })
       className="co-alert"
       title={
         <>
-          {`There are ${alertCount} ${variant} alerts.`}
+          {t('utils~There is {{count}} {{variant}} alert.', { count: alertCount, variant })}
```
Comment (spadgett) [inline frontend/public/components/utils/alerts.tsx:28]: I would use the `public` namespace throughout

```suggestion
          {t('public~There is {{count}} {{variant}} alert.', { count: alertCount, variant })}
```

File: frontend/public/components/utils/log-window.jsx [hunk 6]
```diff
@@ -144,10 +146,14 @@ export class LogWindow extends React.PureComponent {
...
-  bufferFull: PropTypes.bool.isRequired,
-  lines: PropTypes.array.isRequired,
-  linesBehind: PropTypes.number.isRequired,
-  status: PropTypes.string.isRequired,
-  updateStatus: PropTypes.func.isRequired,
-};
+export const LogWindow = withTranslation()(LogWindowWithTranslation);
+
+LogWindow.propTypes =
+  WithTranslation &
```
Comment (spadgett) [inline frontend/public/components/utils/log-window.jsx:152]: Is `WithTranslation` mean to use with `propTypes`? I thought that was for TypeScript types.

File: frontend/public/components/utils/storage-class-dropdown.tsx [hunk 7]
```diff
@@ -24,11 +26,15 @@ export class StorageClassDropdownInner extends React.Component<
...
   UNSAFE_componentWillReceiveProps(nextProps) {
-    const { loaded, loadError, resources } = nextProps;
+    const { loaded, loadError, resources, t } = nextProps;
 
     if (loadError) {
       this.setState({
-        title: <div className="cos-error-title">Error Loading {nextProps.desc}</div>,
+        title: (
+          <div className="cos-error-title">
+            {t('utils~Error Loading {{desc}}', { desc: nextProps.desc })}
```
Comment (spadgett) [inline frontend/public/components/utils/storage-class-dropdown.tsx:35]:
```suggestion
            {t('utils~Error loading {{desc}}', { desc: nextProps.desc })}
```

File: frontend/public/components/utils/log-window.jsx [hunk 8]
```diff
@@ -113,15 +114,16 @@ export class LogWindow extends React.PureComponent {
...
-    const linesBehindCount = pluralize(linesBehind, 'new line');
-    const headerText = bufferFull ? `last ${totalLineCount}` : totalLineCount;
+    const headerText = bufferFull
+      ? t('utils~last {{count}} line', { count: lines.length })
+      : t('utils~{{count}} line', { count: lines.length });
     const resumeText =
-      linesBehind > 0 ? ` Resume stream and show ${linesBehindCount}` : ' Resume stream';
+      linesBehind > 0
+        ? t('utils~ Resume stream and show {{count}} new line', { count: linesBehind })
+        : t('utils~ Resume stream');
```
Comment (spadgett) [inline frontend/public/components/utils/log-window.jsx:126]: We don't want the leading whitespace in the message itself.

**Review by spadgett — APPROVED** [review]
LGTM, please squash. Thanks @jcaianirh !

**Comment by jcaianirh** [comment]
@spadgett squashed and ready to merge

**Review by spadgett — APPROVED** [review]
/lgtm

**Pull request merged**